new/sign-v1.3.md

```diff
@@ -0,0 +1,170 @@
+| Timestamp          | 2009-01-01T12:00:00Z   |
+| User               | spec-user              |
+| Server Api Version | 1                      |
+| Sign Protocol      | v1.3 SHA256            |
+
+Using this information, we can generate the following message:
+
+```
+Method:POST
+Hashed Path:Z3EsTMw/UBNY9n+q+WBWTJmeVg8hQFbdFzVWRxW4dOA=
+X-Ops-Content-Hash:hDlKNZhIhgso3Fs0S0pZwJ0xyBWtR1RBaeHs1DrzOho=
+X-Ops-Sign:algorithm=sha256;version=1.3
+X-Ops-Timestamp:2009-01-01T12:00:00Z
+X-Ops-UserId:/pNOhczwdkQGXD4YAOBVm38wgq2WI7vKRk9d8WBhyKA=
+X-Ops-Server-API-Version:1
+```
+
+Note the line endings are `\n`, and there is no newline after `X-Ops-Server-API-Version:1`.
+
+This message can now be signed, with the expected Base64 representation being:
+
+```
+BjR+iTK2eOgwmT2yGqLvE7Fp+VlpRGyL1dVoF2DmhUPO7EVsnxx2s32AmlOw
+EpaACpav8SoB7K4rpOo3gfBm0XAYLnLLWzcec2OQG2O0wxxHiKVn4qWEe7Cs
+RZ903DGM54t4uK75vx6wwoEdZqZe21npsLK+F3oAqnkgp+YXmlYv9Se5tFKB
+0GWM1ibGJMjUIFAm7vxzjcuEvkkKN49MnXeMAAykfymcs74RU6xEKYzzSAyC
+ygkV6xQSapDMp/aY29cVA/1FgZeVMhnFSTjtqBehchZYwXswr0A72A86gID9
+h2QsUpmQJwbOK3bb1GptAnd5IiLzIxtu+vFeY6h4eA==
+```
+
+The Chef server should expect the following headers to be passed along:
+
+| name                  | value                                                        |
+|-----------------------|--------------------------------------------------------------|
+| X-Ops-Authorization-1 | BjR+iTK2eOgwmT2yGqLvE7Fp+VlpRGyL1dVoF2DmhUPO7EVsnxx2s32AmlOw |
+| X-Ops-Authorization-2 | EpaACpav8SoB7K4rpOo3gfBm0XAYLnLLWzcec2OQG2O0wxxHiKVn4qWEe7Cs |
+| X-Ops-Authorization-3 | RZ903DGM54t4uK75vx6wwoEdZqZe21npsLK+F3oAqnkgp+YXmlYv9Se5tFKB |
+| X-Ops-Authorization-4 | 0GWM1ibGJMjUIFAm7vxzjcuEvkkKN49MnXeMAAykfymcs74RU6xEKYzzSAyC |
+| X-Ops-Authorization-5 | ygkV6xQSapDMp/aY29cVA/1FgZeVMhnFSTjtqBehchZYwXswr0A72A86gID9 |
+| X-Ops-Authorization-6 | h2QsUpmQJwbOK3bb1GptAnd5IiLzIxtu+vFeY6h4eA==                 |
+
+## Compatibility
+The Chef server currently returns the supported protocol versions when failing to authenticate. This information is returned in the `WWW-Authenticate` header of the response along with a response code of `401`. An example of a value that is returned that header is `X-Ops-Sign version="1.0" version="1.1"`, with a  response body of `{"error":["Invalid signature for user or client 'foo'"]}`. The response provides no information as to why the authentication failed.
+
+With servers supporting signing protocol v1.3, `WWW-Authenticate` can signal which hashing algorithms are supported for the 1.3 signing protocol. For example, a server supporting v1.3 could respond with `X-Ops-Sign version="1.0" version="1.1" version="1.3:sha1" version="1.3:sha256"`. This could be the basis of a negotiation protocol that determines which signing algorithm to use. Until such time, Chef client will continue to default to `v1.0`, and using `v1.3` will require setting `:authenticate_protocol_version` in the Chef client configuration to `"1.3"`.
```

Comment:
Either way: leave all discussion of Chef Client's default version out of the RFC, or make a note that all supported servers support 1.1 and we're making that the new default.